app/src/main/java/com/example/levi9application/fragments/RegisterFragment.kt

```diff
@@ -1,7 +1,84 @@
 package com.example.levi9application.fragments
 
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.fragment.findNavController
 import com.example.levi9application.R
+import com.example.levi9application.databinding.FragmentRegistrationBinding
+import com.example.levi9application.viewModels.AuthViewModel
+import dagger.hilt.android.AndroidEntryPoint
 
+@AndroidEntryPoint
 class RegisterFragment: Fragment(R.layout.fragment_registration) {
+    private var _binding: FragmentRegistrationBinding? = null
+    private val binding get() = _binding!!
+    private lateinit var viewModel: AuthViewModel
+
+    private val emailFilter = Regex("[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\\\\.[a-zA-Z]{2,}")
+    private val passwordFilter = Regex("^(?=.*[A-Za-z])(?=.*\\\\d)[A-Za-z\\\\d]{6,}\$")
+    private val nameFilter = Regex("^[a-zA-Z]+$")
+
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentRegistrationBinding.inflate(inflater,container,false)
+        viewModel = ViewModelProvider(this)[AuthViewModel::class.java]
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.btnLogin.setOnClickListener{
+            findNavController().navigate(R.id.action_registerFragment_to_loginFragment)
```

Comment:
We need to clear back stack when navigating from Register to Login fragment. Not doing so will cause the following problem. When we are on Login fragment and we click register button, we will navigate to Register screen. Then, we can click login button to go to Login screen. And if we do these steps 5 times our backstack will be filled with duplicated fragments. Pressing back arrow will cycle back through all the previous fragments which we do not want. Handle this through navigation actions.

Reply:
When navigating from Register to Login fragment app crashes